src/libraries/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.cs

```diff
@@ -32,8 +33,59 @@ public ModuleBuilder DefineDynamicModule(string name)
             return GetDynamicModuleCore(name);
         }
 
+        /// <summary>
+        /// Defines an <see cref="AssemblyBuilder"/> that can be saved to a file or stream.
+        /// </summary>
+        /// <param name="name">The name of the assembly.</param>
+        /// <param name="coreAssembly">The assembly that denotes the "system assembly" that houses the well-known types such as <see cref="object"/></param>
+        /// <param name="assemblyAttributes">A collection that contains the attributes of the assembly.</param>
+        /// <returns>An <see cref="AssemblyBuilder"/> that can be persisted.</returns>
+        /// <exception cref="ArgumentNullException">The <paramref name="name"/> or <paramref name="name.Name"/> or <paramref name="coreAssembly"/> is null.</exception>
+        /// <remarks>Currently the persisted assembly doesn't support running, need to save it and load back to run.</remarks>
+        public static AssemblyBuilder DefinePersistedAssembly(AssemblyName name, Assembly coreAssembly, IEnumerable<CustomAttributeBuilder>? assemblyAttributes = null)
+        {
+            ArgumentNullException.ThrowIfNull(name);
+            ArgumentNullException.ThrowIfNull(coreAssembly);
+
+            Type assemblyType = Type.GetType("System.Reflection.Emit.AssemblyBuilderImpl, System.Reflection.Emit", throwOnError: true)!;
+            ConstructorInfo? con = assemblyType.GetConstructor(BindingFlags.NonPublic | BindingFlags.Instance, [typeof(AssemblyName), typeof(Assembly), typeof(IEnumerable<CustomAttributeBuilder>)]);
+
+            if (con == null)
+            {
+                throw new NullReferenceException(SR.NullReference_AssemblyBuilderConstructor);
+            }
+
```

Comment:
```suggestion
            ConstructorInfo con = assemblyType.GetConstructor(BindingFlags.NonPublic | BindingFlags.Instance, [typeof(AssemblyName), typeof(Assembly), typeof(IEnumerable<CustomAttributeBuilder>)])!;
```
If the method is missing, it means there is something seriously wrong. We do not bother with providing custom error message in other similar places in CoreLib. For example:
https://github.com/dotnet/runtime/blob/59a38f18b777155f01c66afde35f2c0a48850cb0/src/libraries/System.Private.CoreLib/src/System/Reflection/Emit/ILGenerator.cs#L114

Reply:
(If you want to keep the custom exception, it should be `MissingMethodException` constructed with type and method names. Custom message is not needed.)

Reply:
I like MissingMethodException better than NullReferenceException.

Reply:
> If the method is missing, it means there is something seriously wrong. We do not bother with providing custom error message in other similar places in CoreLib. 

Makes sense, thank you!